lib/src/puppetdb-cli.cc

```diff
@@ -107,7 +112,83 @@ PuppetDBConn::parseServerUrls(const json::JsonContainer& config) {
     }
 }
 
-size_t write_data(void *ptr, size_t size, size_t nmemb, FILE *stream) {
+class SynchronizedCharQueue {
```

Comment:
I might be inclined toward a different name; sounds to me like a queue of chars, but it's really a queue of char arrays.  Even SynchronizedCharsQueue might be enough?  Or SynchronizedBufQueue, which suggests the purpose...